// cac-verify/build/controllers/searchCompany.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const utils_1 = require("../utils");
const searchCompany = (query) => __awaiter(void 0, void 0, void 0, function* () {
    const body = {
        searchTerm: query,
    };
    try {
        const data = yield fetch(utils_1.config.PAGE_URL, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                "Access-Control-Allow-Origin": "*",
                'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS'
            },
            body: JSON.stringify(body),
        });
        const response = yield data.json();
        const companies = response.data.forEach(obj => utils_1.serialize(obj));
        return {
            data: companies,
            success: response.success,
        };
    }
    catch (error) {
        return {
            success: false,
            error: 'An error occurred'
        };
    }
});
exports.default = searchCompany;
